inline-caching/README.md

```diff
@@ -0,0 +1,247 @@
+The web server is a Spring Boot web service configured to delegate all read and write operations
+to a GemFire cluster using Spring Boot and Spring Data.  
+
+In our web projects `application.properties` file, 
+we have the following line
+
+```plain
+spring.data.gemfire.pool.locators=localhost[10334]
+```
+This tells the GemFire client the location of our GemFire cluster that we created above. 
+The port `10334` is the default port, but any port can be used if configured to do so.
+
+Starting the spring boot app with GemFire can be done via
+
+```bash
+./gradlew bootRun
+```
+
+When the application starts, you should see in the Tomcat log files that the GemFire client has discovered a locator.
+
+```plain
+AutoConnectionSource discovered new locators [...:10334]
+```
+
+This means that our Spring Boot web application is connected and ready to handle requests. 
+
+## Performing Requests
+
+**The following web requests can be made to test the Cache Loader**
+
+Add a value to postgres database. This will bypass GemFire but will change a value in postgres.
+
+    psql postgres -U myuser;
+    update pgbench_tellers set filler = 'hello' where tid = 1;
+
+Retrieve value from webservice. This will invoke GemFire's CacheLoader. GemFire shouldn't know of this 
+data with the cooresponding key `1` and therefore will reach out to postgres.
+
+    curl localhost:8080/1          # should see the value hello
+
+Change value on the postgres database. Again, this will bypass GemFire but will update the value in the backing database. 
+This will setup a demonstration that GemFire is indeed caching values. 
+
+    psql postgres -U myuser;
+    update pgbench_tellers set filler = 'goodbye' where tid = 1;
+
+Retrieve value from webservice. This will still show the originally set value because the web service / GemFire was not involved. 
+
+    curl localhost:8080/1          # should still see the value hello because it is cached
+
+Clear the cache in GemFire
+
+    remove --key=1 --region=item
+
+Retrieve the value from the webservice after clearing the GemFire cache so that the CacheLoader will be invoked again. 
+
+    curl localhost:8080/1          # should see the new value goodbye
+
+**The following web requests can be made to test the Async Event Listener**
+
+Write value to webservice. This will write a value to GemFire first, and then the AsyncEventListener will pick up the write event and 
+persist the value to postgres. 
+
+    curl -X PUT localhost:8080/1 -H 'Content-Type: application/json' -d '{"value":"potato"}'
+
+See value in postgres. We can see here that the value has been written. 
+
+    psql postgres -U myuser;
+    select filler from pgbench_tellers where tid=1;  -- should see potato
+
+Retrieve value from webservice. Just to be certain, we can see the new value has been updated for reads as well. 
+The AsyncEventListener and CacheLoader work together. 
+
+    curl localhost:8080/1          # should see the value potato
+
```

Comment:
We need to include a clean up/delete/shutdown section that shutdowns and cleans up all the things we have running.

Reply:
added cleanup section